src/Domain/CampaignStatisticsRepository.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace MatchBot\Domain;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\EntityRepository;
+
+class CampaignStatisticsRepository
+{
+    /** @var EntityRepository<CampaignStatistics>  */
+    private EntityRepository $doctrineRepository;
+
+    /**
+     * @psalm-suppress PossiblyUnusedMethod Called by DI container
+     */
+    public function __construct(private EntityManagerInterface $em)
+    {
+        $this->doctrineRepository = $em->getRepository(CampaignStatistics::class);
+    }
+
+    /**
+     * Creates or finds + updates a {@see CampaignStatistics} record; doesn't flush, so callers need to when done
+     * building stats.
+     */
+    public function updateStatistics(Campaign $campaign, Money $amountRaised, Money $matchFundsUsed): void
+    {
+        $statistics = $this->doctrineRepository->findOneBy(['campaign' => $campaign]);
+
+        if ($statistics) {
+            $statistics->setAmountRaised($amountRaised);
+            $statistics->setMatchFundsUsed($matchFundsUsed);
+        } else {
+            $statistics = new CampaignStatistics($campaign, $amountRaised, $matchFundsUsed);
+            $this->em->persist($statistics);
+        }
+    }
+
+    /**
+     * With figures all zero if no donations or match funds used yet *or* if first stats calculation
+     * has not yet run.
+     */
+    public function getStatistics(Campaign $campaign): CampaignStatistics
+    {
+        $statistics = $this->doctrineRepository->findOneBy(['campaign' => $campaign]);
+
+        if (!$statistics) {
+            return new CampaignStatistics($campaign, Money::zero(), Money::zero());
```

Comment:
can we return zeroes of the currency from the campaign instead of zero pounds? Maybe we shouldn't have a default currency on the zero function since its too easy to forget to specify a currency here.